Plugin/Model/Order/ItemRepositoryPlugin.php

```diff
@@ -0,0 +1,179 @@
+        SerializerInterface $serializer
+    ) {
+        $this->orderItemExtensionFactory = $orderItemExtensionFactory;
+        $this->sourceSelectionResult = $sourceSelectionResult;
+        $this->getInventoryRequestFromOrder = $getInventoryRequestFromOrder;
+        $this->getDefaultSourceSelectionAlgorithmCode = $getDefaultSourceSelectionAlgorithmCode;
+        $this->sourceSelectionService = $sourceSelectionService;
+        $this->serializer = $serializer;
+    }
+
+    /**
+     * Set additional attributes when requesting a single order item
+     *
+     * @param ItemRepository  $subject
+     * @param OrderItemInterface|Item $result
+     * @return OrderItemInterface|Item
+     */
+    public function afterGet(ItemRepository $subject, OrderItemInterface $result)
+    {
+        return $this->applyExtensionAttributesToOrderItem($result, $result);
+    }
+
+    /**
+     * Set additional attributes when requesting a list of order items
+     *
+     * @param ItemRepository $subject
+     * @param OrderItemSearchResultInterface $result
+     * @return OrderItemSearchResultInterface
+     */
+    public function afterGetList(ItemRepository $subject, OrderItemSearchResultInterface $result)
+    {
+        foreach ($result->getItems() as $item) {
+            $this->applyExtensionAttributesToOrderItem($item, $result->getItems());
+        }
+
+        return $result;
+    }
+
+    /**
+     * Apply extension attributes to order item
+     *
+     * @param OrderItemInterface $orderItem
+     * @param array $allItems
+     * @return OrderItemInterface
+     */
+    private function applyExtensionAttributesToOrderItem(OrderItemInterface $orderItem, array $allItems): OrderItemInterface
+    {
+        if (!$extensionAttributes = $orderItem->getExtensionAttributes()) {
+            $extensionAttributes = $this->orderItemExtensionFactory->create();
+        }
+
+        $extensionAttributes->setSources(
+            $this->serializer->serialize(
+                $this->getOrderItemSources($orderItem, $allItems)
+            )
+        );
+
+        $orderItem->setExtensionAttributes($extensionAttributes);
+
+        return $orderItem;
+    }
+
+    /**
+     * @param OrderItemInterface $orderItem
+     * @param array $allItems
+     * @return array
+     */
+    private function getOrderItemSources(OrderItemInterface $orderItem, array $allItems): array
+    {
+        if ($this->sourceSelectionItems === null) {
+            $inventoryRequest = $this->getInventoryRequestFromOrder->execute(
+                $orderItem->getOrder(),
+                $this->sourceSelectionResult->getSelectionRequestItems($allItems)
+            );
+
+            $selectionAlgorithmCode = $this->getDefaultSourceSelectionAlgorithmCode->execute();
+            $sourceSelectionResult = $this->sourceSelectionService->execute($inventoryRequest, $selectionAlgorithmCode);
+
+            $this->sourceSelectionItems = $sourceSelectionResult->getSourceSelectionItems();
```

Comment:
Hey @ahasAmp couldn't we iterate the `sourceSelectionItems` here, set the source data and index by item sku? We could then return the data based on the `$orderItem->getSku()`, for example:
```php
<?php

use Magento\InventorySourceSelectionApi\Api\Data\SourceSelectionItemInterface;

$orderItemSku = $orderItem->getSku();
if (array_key_exists($orderItemSku, $this->sourceSelectionItems)) {
    return $this->sourceSelectionItems[$orderItemSku];
}

...

$this->sourceSelectionItems = array_reduce(
    $sourceSelectionResult->getSourceSelectionItems(),
    function (array $indexedArray, SourceSelectionItemInterface $selectionItem) {
        $indexedArray[$selectionItem->getSku()] = [
            'source_code' => $item->getSourceCode(),
            'qty' => $item->getQtyToDeduct()
        ];

        return $indexedArray;
    },
    []
);
```
Or something to that effect. There's likely a better way of indexing them initially so we wouldn't have to recreate the entire array if a certain SKU doesn't exist